src/UI/home/dashboard/chart.tsx

```diff
@@ -1,20 +1,51 @@
-import {PieChart} from "react-minimal-pie-chart";
-import React from "react";
-import {cryptoChart} from "../types";
+import {PieChart} from 'react-minimal-pie-chart'
+import React, {useEffect, useState} from 'react'
+import {cryptoChart} from '../types'
+import {CoinInWallet} from '../../../backend-service-connector/model/rest'
 
 export function Chart(props: any) {
-    const userService = props.userService
+    const userWallet = props.userWallet
+    const allUserCoins: Array<CoinInWallet> = userWallet.allCoinsInWallet
+    const [color, setColor] = useState<Array<any>>([])
+    let randomColor = require('randomcolor')
 
-    const data: Array<cryptoChart> = userService.getDataForChart()
+    useEffect(() => {
+        let arrayOfColors: Array<any> = []
+
+        for (let i = 0; i < allUserCoins.length; i++) {
+            const color = randomColor()
+
+            arrayOfColors = arrayOfColors.concat(color)
+        }
+
+        setColor(arrayOfColors)
+    }, [])
 
     const renderCryptoNames = () => {
-        return data.map((crypto, index) => {
-            return (
-                <CryptoInChart key={index} color={crypto.color} title={crypto.title}/>
-            )
+
+        return allUserCoins.map((crypto, index) => {
+            if (crypto.walletPercent !== 0.00) {
+                return (
+                    <CryptoInChart
+                        key={index}
+                        color={color[index]}
+                        title={`${crypto.coinUserDto.name} ${(Math.round(crypto.walletPercent * 100) / 100).toFixed(2)}%`}
+                    />
+                )
+            }
         })
     }
 
+    const data: Array<cryptoChart> = allUserCoins.map((coin, index) => {
+        const chartData: cryptoChart = {
+            color: color[index],
+            title: coin.coinUserDto.name,
+            value: coin.walletPercent
+        }
+
+        return chartData
```

Comment:
you can shorten it with `allUserCoins.map((coin, index) => ({
 color: color[index],
 title: coin.coinUserDto.name,
 value: coin.walletPercent
}))
`